portfolio/src/main/webapp/nav.js

```diff
@@ -0,0 +1,21 @@
+/* change src of iframe according to input index
+*/
+function load(str) {
+  var frame = document.getElementsByTagName("iframe")[0];
+  switch(str) {
+    case 1:
+      frame.src = "main.html";
+      break;
+    case 2:
+      frame.src = "introduction.html";
+      break;
+    case 3:
+      frame.src = "projects.html";
+      break;
+    case 4:
+      frame.src = "gallery.html";
+      break;
+    default:
+      frame.src = "error.html";
+  }
```

Comment:
While this definitely works, it would be simpler to create an array of pages. Just make sure you add checks that the index is valid.

```
frame.src = navigateablePages[index];
```

Reply:
+1 Also, use const here instead of let.